src/osd/osd_types.cc

```diff
@@ -6312,6 +6318,59 @@ void object_info_t::decode(ceph::buffer::list::const_iterator& bl)
   DECODE_FINISH(bl);
 }
 
```

Comment:
I'd rather we clear the oid field and use the existing encode/decode calls.  Doing it this way means that we'll crash when decoding an object_info_t encoded by classic.

Medium term, we probably want to simply remove the oid field from both classic and crimson and update the encoding version.

Reply:
That is, encode_no_oid would swap out the oid field, encode, and then restore it.

Reply:
Fixed.